### DISCUSSION TITLE
Release 2.0.0b4 - beta quality, for testing only

### DISCUSSION THREAD

**Discussion by ThomasWaldmann** [borgbackup/borg#7168] [General]
borgbackup 2.0.0b4 beta release - do not use this on production backup repositories.

Long changelog:
https://borgbackup.readthedocs.io/en/2.0.0b4/changes.html#version-2-0-0b4-2022-11-27

If you use pip to install this, use: pip install "borgbackup==2.0.0b4"

Docs: https://borgbackup.readthedocs.io/en/2.0.0b4/

Note: no macOS binaries yet, they maybe come later.

Feedback:

<hr /><em>This discussion was created from the release <a href='https://github.com/borgbackup/borg/releases/tag/2.0.0b4'>Release 2.0.0b4 - beta quality, for testing only</a>.</em>

**Comment by m3nu** [comment]
Just installed on all [BorgBase](https://www.borgbase.com/) boxes. No issues with installation at least. Will get to testing from CLI and Vorta tomorrow.

**Comment by m3nu** [comment]
Finished some benchmarks comparing 1.2 to 2.0. Key findings:

- Borg 2.0 with Python 3.11 is about 5 to 20% faster on normal- and high latency connections and about the same memory. The difference was smaller, when the focus was on compression.
- CPU time is drastically lower for Borg 2.0. As much as 30%, unless more compression is involved.
- Final backup size was the same for Borg 1.2 and 2.0

I also compared Restic and some network optimizations people keep bringing up. Full results and script here: https://github.com/borgbase/benchmarks

**Reply by Arcitec** [reply]
Thank you for testing and providing benchmarks. Very interesting! These are the exact reasons I chose Borg instead of Restic, and I'm glad to see Borg 2.0 will be even faster. :)

**Reply by infectormp** [reply]
Maybe just try compiling borg with [codon](https://docs.exaloop.io/codon) as an experiment? and check performance results again.

**Comment by m3nu** [comment]
Was wondering: Could the repo encryption option have a sensible default or short alias instead of

`borg rcreate --encryption=repokey-blake2-chacha20-poly1305`

this as alias to the current recommended option:

`borg rcreate --encryption=repokey` ?

The latter is short enough to remember. The former needs copy & pasting always. The docs refer to `borg benchmark cpu`.  So it could pick the best algo based on a short benchmark?

Impact of choosing the wrong thing is pretty significant. 4x in my benchmark for "encryption", 2x for hashes.

The user only really needs to pick repokey or keyfile.

```
Encryption =====================================================
aes-256-ctr-hmac-sha256  1GB        21.289s
aes-256-ctr-blake2b      1GB        18.172s
aes-256-ocb              1GB        13.424s
chacha20-poly1305        1GB        4.684s
```

**Reply by ThomasWaldmann** [reply]
I thought about this, noticing that the quite long names are long...

But I don't want to just have a default here, for the same reasons we did not have a default here before: people need to make an informed decision here, based on the benchmark outcome and the docs.

Having short aliases also doesn't help. Basically it then would need a table, listing the alias and what it really means. And you'ld still have to find out what's best for you. So, if you have to look it up in the docs anyway, you also can c&p the long name from there and we can save implementing/documenting a "nothing saying" short name / alias for that.

It's also relevant for creation of related repos, people should be well aware what id algorithm they have.

About the benchmark you posted:
Usually aes-256-ocb should be faster (if AES is hw accelerated). Was that from a VM?

**Reply by m3nu** [reply]
Got it. Good to know the background.

Yes, the test was from a VM.
